pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -232,6 +245,35 @@ def _configure_fault_tolerance_callbacks(self):
         # don't use `log_dir` to minimize the chances of failure
         self.trainer.callbacks.append(_FaultToleranceCheckpoint(dirpath=self.trainer.default_root_dir))
 
+    def _configure_external_callbacks(self) -> None:
+        """Add external callbacks registered through entry points.
+
+        The entry points are expected to be functions returning a list of callbacks, which will be added to the Trainer
+        callback list.
+        """
+        if _PYTHON_GREATER_EQUAL_3_8_0:
+            factories = importlib.metadata.entry_points().get("pytorch_lightning.callbacks_factory", ())
```

Comment:
nit: below you use the `List[Callback]` type, so this should be 

```suggestion
            factories = importlib.metadata.entry_points().get("pytorch_lightning.callbacks_factory", [])
```

Reply:
yes but that type annotation is not for the same. I chose tuple here because the get() method here returns a tuple normally.

Reply:
the for loop type check should be happy with any iterable